runtime/rococo/src/bridge_messages.rs

```diff
@@ -0,0 +1,409 @@
+/// The message must have dispatch weight set to zero. The result then must be rounded up to account
+/// possible future runtime upgrades.
+const DEFAULT_MESSAGE_DELIVERY_TX_WEIGHT: Weight = 1_500_000_000;
+/// Weight of pay-dispatch-fee operation for inbound messages at Rococo/Wococo chain.
+///
+/// This value corresponds to the result of `pallet_bridge_messages::WeightInfoExt::pay_inbound_dispatch_fee_overhead()`
+/// call for your chain. Don't put too much reserve there, because it is used to **decrease**
+/// `DEFAULT_MESSAGE_DELIVERY_TX_WEIGHT` cost. So putting large reserve would make delivery transactions cheaper.
+const PAY_INBOUND_DISPATCH_FEE_WEIGHT: Weight = 600_000_000;
+/// Number of bytes, included in the signed Rococo/Wococo transaction apart from the encoded call itself.
+///
+/// Can be computed by subtracting encoded call size from raw transaction size.
+const TX_EXTRA_BYTES: u32 = 130;
+
+/// Rococo chain as it is seen at Rococo.
+pub type RococoAtRococo = RococoLikeChain<AtRococoWithWococoMessageBridge, crate::RococoGrandpaInstance>;
+
+/// Rococo chain as it is seen at Wococo.
+pub type RococoAtWococo = RococoLikeChain<AtWococoWithRococoMessageBridge, crate::RococoGrandpaInstance>;
+
+/// Wococo chain as it is seen at Wococo.
+pub type WococoAtWococo = RococoLikeChain<AtWococoWithRococoMessageBridge, crate::WococoGrandpaInstance>;
+
+/// Wococo chain as it is seen at Rococo.
+pub type WococoAtRococo = RococoLikeChain<AtRococoWithWococoMessageBridge, crate::WococoGrandpaInstance>;
+
+/// Rococo/Wococo chain from message lane point of view.
+#[derive(RuntimeDebug, Clone, Copy)]
+pub struct RococoLikeChain<B, GI> {
+	__bridge_definition: PhantomData<B>,
+	_at_this_chain_grandpa_pallet_instance: PhantomData<GI>,
+}
+
+impl<B, GI> ChainWithMessages for RococoLikeChain<B, GI> {
+	type Hash = crate::Hash;
+	type AccountId = crate::AccountId;
+	type Signer = primitives::v1::AccountPublic;
+	type Signature = crate::Signature;
+	type Weight = Weight;
+	type Balance = crate::Balance;
+}
+
+impl<B, GI> ThisChainWithMessages for RococoLikeChain<B, GI> {
+	type Call = crate::Call;
+
+	fn is_outbound_lane_enabled(lane: &LaneId) -> bool {
+		*lane == [0, 0, 0, 0]
+	}
+
+	fn maximal_pending_messages_at_outbound_lane() -> MessageNonce {
+		MAXIMAL_PENDING_MESSAGES_AT_OUTBOUND_LANE
+	}
+
+	fn estimate_delivery_confirmation_transaction() -> MessageTransaction<Weight> {
+		let inbound_data_size = InboundLaneData::<crate::AccountId>::encoded_size_hint(
+			MAXIMAL_ENCODED_ACCOUNT_ID_SIZE,
+			1,
+			1,
+		)
+		.unwrap_or(u32::MAX);
+
+		MessageTransaction {
+			dispatch_weight: MAX_SINGLE_MESSAGE_DELIVERY_CONFIRMATION_TX_WEIGHT,
+			size: inbound_data_size
+				.saturating_add(EXTRA_STORAGE_PROOF_SIZE)
+				.saturating_add(TX_EXTRA_BYTES),
+		}
+	}
+
+	fn transaction_payment(transaction: MessageTransaction<Weight>) -> crate::Balance {
+		// current fee multiplier is used here
```

Comment:
This will also likely be changed (e.g. x2-ed, not sure yet) in the P<>K deployment, because it describes future transactions, where multiplier may increase.